### PR TITLE
feat(query): support implicit workspace

### PR DESCRIPTION
**Pull request by boyapatichandu**
**Pull Request checklist**

- [ ] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** Queries without `_ws_` key fails while creating exec plan

`foo{_ns_="foo"}` fails with bad request

**New behavior :** Enable ShardKeyRegexPlanner to utilize shardKeyMatcher columnFilters to derive the missing `_ws_` key

`foo{_ns_="foo"}` passes if `_ws_` info is available in shardKeyMatcher.

### CODE REVIEW

**Review by tjackpaul — APPROVED** [review]
lgtm.. thanks for the PR @boyapatichandu

File: coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala [hunk 1]
```diff
@@ -58,14 +58,25 @@ class ShardKeyRegexPlanner(val dataset: Dataset,
...
   }
 
+  def hasRequiredShardKeysPresent(nonMetricShardKeyFilters: Seq[Seq[ColumnFilter]],
+                                  nonMetricShardColumns: Seq[String]): Boolean = {
+    nonMetricShardKeyFilters
+      .foreach(filterGroup => {
+        val columnNames = filterGroup.map(_.column)
+        if (!nonMetricShardColumns.toSet.subsetOf(columnNames.toSet)) return false
+      })
+    true
```
Comment (alextheimer) [inline coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala:77]: Does this always return `true`? Could you add a "should _not_ generate EP for implicit ws query" test?

File: coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala [hunk 2]
```diff
@@ -58,14 +58,25 @@ class ShardKeyRegexPlanner(val dataset: Dataset,
     val nonMetricShardKeyFilters =
       LogicalPlan.getNonMetricShardKeyFilters(logicalPlan, dataset.options.nonMetricShardColumns)
     if (isMetadataQuery(logicalPlan)
-      || LogicalPlan.hasShardKeyEqualsOnly(logicalPlan, dataset.options.nonMetricShardColumns)) {
+      || (hasRequiredShardKeysPresent(nonMetricShardKeyFilters, dataset.options.nonMetricShardColumns) &&
```
Comment (kvpetrov) [inline coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala:61]: difficult to follow the logic
if (A || B && C)
 
 (a || B ) && C?
or 
A || (B && C) ?

Comment (boyapatichandu) [reply coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala:61]: >A || (B && C) ?
if its metadata query OR (has all the required nonMetric ShardKeys (`_ws_`, `_ns_`) present with Equals filter )

File: coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala (hunk 1 above)
Comment (boyapatichandu) [reply coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala:77]: Depends on the query. If query has both nonMetricShardkeys `_ws_` and `_ns_` returns true(e.x `metric{_ws_:foo, _ns_:bar}`) Else false (e.x `metric{ _ns_:bar}`)

File: coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala [hunk 3]
```diff
@@ -58,14 +58,25 @@ class ShardKeyRegexPlanner(val dataset: Dataset,
...
       generateExecWithoutRegex(logicalPlan, nonMetricShardKeyFilters.head, qContext).head
     } else walkLogicalPlanTree(logicalPlan, qContext).plans.head
   }
 
+  def hasRequiredShardKeysPresent(nonMetricShardKeyFilters: Seq[Seq[ColumnFilter]],
+                                  nonMetricShardColumns: Seq[String]): Boolean = {
+    nonMetricShardKeyFilters
+      .foreach(filterGroup => {
+        val columnNames = filterGroup.map(_.column)
+        if (!nonMetricShardColumns.toSet.subsetOf(columnNames.toSet)) return false
```
Comment (amolnayak311) [inline coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala:75]:
```
  def hasRequiredShardKeysPresent(nonMetricShardKeyFilters: Seq[Seq[ColumnFilter]],
                                  nonMetricShardColumns: Seq[String]): Boolean = {
    val nonMetricShardColumnsSet = nonMetricShardColumns.toSet
    nonMetricShardKeyFilters
      .foreach(filterGroup => 
            nonMetricShardColumnsSet.subsetOf(filterGroup.map(_.column).toSet)
        )    
  }
```

**Review by alextheimer — APPROVED** [review]
Previous comment resolved-- lgtm!

**Review by amolnayak311 — APPROVED** [review]
Minor changes suggested

File: coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala (hunk 2 above)
Comment (kvpetrov) [reply coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala:61]: got you, got you!

**Pull request merged**